range.js

```diff
@@ -44,63 +44,129 @@ $.fn.range = function(parameters) {
 				
 				inner,
 				thumb,
+				rthumb,
+				thumbpos,
+				rthumbpos,
+				thumbval,
+				rthumbval,
 				trackLeft,
 				precision,
-				
 				module
 			;
-			
 			module = {
 				
 				initialize: function() {
-					module.instantiate();
 					module.sanitize();
+					module.instantiate();
+					$(window).resize(function() {
+						module.instantiate();
+					});
 				},
 				
 				instantiate: function() {
 					instance = module;
 					$module
 						.data(moduleNamespace, module)
 					;
-					$(element).html("<div class='inner'><div class='track'></div><div class='track-fill'></div><div class='thumb'></div></div>");
-					inner = $(element).children('.inner')[0];
-					thumb = $(element).find('.thumb')[0];
-					trackLeft = $(element).find('.track-fill')[0];
-					// find precision of step, used in calculating the value
-					module.determinePrecision();
-					// set start location
-					module.setValuePosition(settings.start);
-					// event listeners
-					$(element).find('.track, .thumb, .inner').on('mousedown', function(event) {
-						event.stopImmediatePropagation();
-						event.preventDefault();
-						$(this).closest(".range").trigger('mousedown', event);
-					});
-					$(element).find('.track, .thumb, .inner').on('touchstart', function(event) {
-						event.stopImmediatePropagation();
-						event.preventDefault();
-						$(this).closest(".range").trigger('touchstart', event);
-					});
-					$(element).on('mousedown', function(event, originalEvent) {
-						module.rangeMousedown(event, false, originalEvent);
-					});
-					$(element).on('touchstart', function(event, originalEvent) {
-						module.rangeMousedown(event, true, originalEvent);
-					});
+					if (settings.values) {
+						$(element).html("<div class='inner'><div class='track'></div><div class='track-fill'></div><div class='thumb'></div><div class='thumb'></div></div>");
+						inner = $(element).children('.inner')[0];
+						thumb = $(element).find('.thumb')[0];
+						rthumb = $(element).find('.thumb')[1];
+						trackLeft = $(element).find('.track-fill')[0];
+						// find precision of step, used in calculating the value
+						module.determinePrecision();
+						// set start location
+						if (settings.start.length) {
+							if (settings.start[0] >= settings.values[0] && settings.start[0] <= settings.values[1]
+								&& settings.start[1] >= settings.values[0] && settings.start[1] <= settings.values[1]) {
+								thumbval = settings.start[0];
+								rthumbval = settings.start[1];
+							} else {
+								thumbval = settings.values[0];
+								rthumbval = settings.values[1];
+							}
+						}
+						settings.min = settings.values[0];
+						settings.max = settings.values[1];	
```

Comment:
Is there a reason to override `min` and `max` here? Why change it to use items from the `values` array?